Formula/trojan-go.rb

```diff
@@ -0,0 +1,114 @@
+class TrojanGo < Formula
+  desc "Trojan proxy in Go"
+  homepage "https://p4gefau1t.github.io/trojan-go/"
+  url "https://github.com/p4gefau1t/trojan-go.git",
+      tag:      "v0.10.0",
+      revision: "a9ce6d4ada9143d58e8d1373d23f9ebfe871119e"
+  license "GPL-3.0-only"
+  head "https://github.com/p4gefau1t/trojan-go.git"
+
+  depends_on "go" => :build
+
+  resource "geoip" do
+    geoip_hash_url = "https://github.com/v2fly/geoip/releases/latest/download/geoip.dat.sha256sum"
+    url "https://github.com/v2fly/geoip/releases/latest/download/geoip.dat"
+    sha256 Utils.safe_popen_read("curl", "-L", geoip_hash_url)[0, 64]
+  end
+
+  resource "geosite" do
+    geosite_hash_url = "https://github.com/v2fly/domain-list-community/releases/latest/download/dlc.dat.sha256sum"
+    url "https://github.com/v2fly/domain-list-community/releases/latest/download/dlc.dat"
+    sha256 Utils.safe_popen_read("curl", "-L", geosite_hash_url)[0, 64]
+  end
+
+  def install
+    commit = Utils.git_head
+    tag = Utils.safe_popen_read("git", "tag", "--contains", "HEAD").strip
+    execpath = libexec / name
+    ldflags = %W[
+      -X github.com/p4gefau1t/trojan-go/constant.Version=#{tag}
+      -X github.com/p4gefau1t/trojan-go/constant.Commit=#{commit}
+    ]
+
+    system "go", "build", *std_go_args, "-o", execpath, "-tags=full", "-ldflags", ldflags.join(" ")
+    (bin / "trojan-go").write_env_script execpath,
+      TROJAN_GO_LOCATION_ASSET: "${TROJAN_GO_LOCATION_ASSET:-#{pkgshare}}"
+
+    pkgetc.install "example/client.json" => "config.json"
+
+    resource("geoip").stage do
+      pkgshare.install "geoip.dat"
+    end
+
+    resource("geosite").stage do
+      pkgshare.install "dlc.dat" => "geosite.dat"
+    end
+  end
+
+  plist_options manual: "trojan-go -config #{HOMEBREW_PREFIX}/etc/trojan-go/config.json"
+
+  def plist
+    <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+        <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+        <plist version="1.0">
+        <dict>
+          <key>Label</key>
+          <string>#{plist_name}</string>
+          <key>ProgramArguments</key>
+          <array>
+            <string>#{bin}/trojan-go</string>
+            <string>-config</string>
+            <string>#{etc}/trojan-go/config.json</string>
+          </array>
+          <key>KeepAlive</key>
+          <true/>
+          <key>RunAtLoad</key>
+          <true/>
+        </dict>
+      </plist>
+    EOS
+  end
```

Comment:
Can we use the `service` DSL here? Example:

https://github.com/Homebrew/homebrew-core/blob/b4e38c605e14bef7c2298e092deb8bdc1813394d/Formula/php.rb#L323-L330

Reply:
I modified it. Now there is only a small difference from the previous:

previous
```text
==> Caveats
To have launchd start trojan-go now and restart at login:
  brew services start trojan-go
Or, if you don't want/need a background service you can just run:
  trojan-go -config /usr/local/etc/trojan-go/config.json
```

now
```text
==> Caveats
To start trojan-go:
  brew services start trojan-go
Or, if you don't want/need a background service you can just run:
  /usr/local/opt/trojan-go/bin/trojan-go -config /usr/local/etc/trojan-go/config.json
```